ibm/resource_ibm_cdn.go

```diff
@@ -0,0 +1,579 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"certificatetype": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "0",
+			},
+			"cachekeyqueryrule": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "include-all",
+			},
+			"performanceconfiguration": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "General web delivery",
+			},
+			"path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "/",
+			},
+		},
+	}
+}
+
+func resourceIBMCDNCreate(d *schema.ResourceData, meta interface{}) error {
+	///create  session
+	sess := meta.(ClientSession).SoftLayerSession()
+	log.Println("ordering cdn service...")
+	///get the value of all the parameters
+	domain := d.Get("hostname").(string)
+	vendorname := d.Get("vendor_name").(string)
+	origintype := d.Get("origin_type").(string)
+	originaddress := d.Get("origin_address").(string)
+	protocol := d.Get("protocol").(string)
+	httpport := d.Get("httpport").(int)
+	httpsport := d.Get("httpsport").(int)
+	bucketname := d.Get("bucketname").(string)
+	path := d.Get("path").(string)
+	header := d.Get("header").(string)
+	cachekeyqueryrule := d.Get("cachekeyqueryrule").(string)
+	performanceconfiguration := d.Get("performanceconfiguration").(string)
+	respectheaders := d.Get("respectheaders").(bool)
+	cname := d.Get("cname").(string)
+	certificateType := d.Get("certificatetype").(string)
+	if cname != "0" {
+		cname = cname + str
+	}
+	if cname == "0" {
+		cname = ""
+	}
+	///creat an object of CDN service
+	service := services.GetNetworkCdnMarketplaceConfigurationMappingService(sess)
+	//////pass the parameters to create domain mapping
+	if origintype == "OBJECT_STORAGE" && protocol == "HTTP" {
+		receipt1, err := service.CreateDomainMapping(&datatypes.Container_Network_CdnMarketplace_Configuration_Input{
+			Origin:                   sl.String(originaddress),
+			VendorName:               sl.String(vendorname),
+			Domain:                   sl.String(domain),
+			Path:                     sl.String(path),
+			Cname:                    sl.String(cname),
+			Protocol:                 sl.String(protocol),
+			HttpPort:                 sl.Int(httpport),
+			OriginType:               sl.String(origintype),
+			BucketName:               sl.String(bucketname),
+			Header:                   sl.String(header),
+			RespectHeaders:           sl.Bool(respectheaders),
+			CacheKeyQueryRule:        sl.String(cachekeyqueryrule),
+			PerformanceConfiguration: sl.String(performanceconfiguration),
+		})
+		if err != nil {
+			return fmt.Errorf("Error creating CDN: %s", err)
+		}
+		///Print the response of the requested the service.
+		log.Print("Response for cdn order")
+		log.Println(receipt1)
+		d.SetId(*receipt1[0].UniqueId)
+		id, err := strconv.Atoi((d.Id()))
+		result1, err := service.VerifyDomainMapping(&id)
+		log.Println(result1)
```

Comment:
Improve the log stmts which are important with proper string like the 
log.Println("The status of domain mapping %s", result.Status)